src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -149,8 +113,8 @@ public override unsafe string ToString()
         public void WriteTo(ObjectWriter writer)
             => _checksum.WriteTo(writer);
 
-        public void WriteTo(Span<byte> span)
-            => _checksum.WriteTo(span);
+        public bool TryWriteTo(Span<byte> span)
+            => MemoryMarshal.TryWrite(span, ref Unsafe.AsRef(in _checksum));
```

Comment:
❔ Since this is an internal API and we don't expect incorrectly-sized spans, would it be better to keep it as `WriteTo`?

```suggestion
        public void WriteTo(Span<byte> span)
            => Contract.ThrowIfFalse(MemoryMarshal.TryWrite(span, ref Unsafe.AsRef(in _checksum)));
```

Reply:
wfm.